pkg/octavia/dbsync.go

```diff
@@ -83,7 +83,7 @@ func DbSyncJob(
 	initContainerDetails := APIDetails{
 		ContainerImage:       instance.Spec.OctaviaAPI.ContainerImage,
 		DatabaseHost:         instance.Status.DatabaseHostname,
-		DatabaseUser:         instance.Spec.DatabaseUser,
+		DatabaseAccount:      instance.Spec.DatabaseAccount,
```

Comment:
this part is going to have to change more.   The prior use of DatabaseUser was also not appropriate here as this forms part of the job hash. it means as the username changes, the hash of the dbsync job changes and it will keep running again.   This hash should only be based on the database name, not the username.

Yesterday I figured all this out and made these changes for designate and heat; the database account / secret has to be put into the configmaps, which also ideally needs to be a secret, not a configmap.

see the designate / heat PRs for how I changed the job hash for dbsync and others to not use account: 
https://github.com/openstack-k8s-operators/designate-operator/pull/156 https://github.com/openstack-k8s-operators/heat-operator/pull/322 

these also change from configmap to secret, based on guidance in this doc:   https://github.com/openstack-k8s-operators/docs/blob/main/service_config.md#proposed-approach 

I took a look at the keystone-operator to see what APIs they use to use a Secret for the config

Reply:
thread:  https://redhat-internal.slack.com/archives/CQXJFGMK6/p1708453474221679

Reply:
yup these still have to come out from here

Reply:
I  also replaced ConfigMapVolumeSource with SecretVolumeSource. I see in the Designate PR that you removed DBPasswordSelector as well, so I removed it here too.